File: 1967-number-of-strings-that-appear-as-substrings-in-word/1967-number-of-strings-that-appear-as-substrings-in-word.py
class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        count=0
        for cWord in patterns:
            found=False
            for start in range(len(word)):
                if found:
                    break
                for end in range(start+1,len(word)+1):
                    if word[start:end]==cWord:
                        count+=1
                        found=True
                        break
        return count
#TC=O(N^3)
#SC=O(1)

File: 1967-number-of-strings-that-appear-as-substrings-in-word/1967-number-of-strings-that-appear-as-substrings-in-word.java
class Solution {
    public int numOfStrings(String[] patterns, String word) {
        int count=0;
        for (int i=0;i<patterns.length;i+=1){
            Boolean found=false;
            String cWord=patterns[i];
            for (int start=0;start<word.length() && !found;start+=1){
                String temp="";
                temp+=word.charAt(start);
                if (temp.equals(cWord)){
                    count+=1;
                    found=true;
                }
                for(int end=start+1;end<word.length();end+=1){

                    temp+=word.charAt(end);
                    if (temp.equals(cWord)){
                        found=true;
                        System.out.println(temp);
                        count+=1;
                        break;
                    }
                    else{
                        continue;
                    }
                }
            }
        }
        return count;
    }
}

//TC=O(N^2)
//SC=O(1)